x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -68,139 +41,201 @@ func RandMsgCreatePosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.
+	if err != nil {
+		return nil, err
 	}
 
-	positions, err := k.GetUserPositions(ctx, sender.Address, 0)
-	if err != nil {
-		return nil, fmt.Errorf("position does not exist")
+	if len(positionIds) < 1 {
+		return nil, fmt.Errorf("user doesnot have any positions")
 	}
 
-	if len(positions) == 0 {
-		return nil, fmt.Errorf("user does not have any position")
+	randPositionId := positionIds[rand.Intn(len(positionIds))]
+	position, err := k.GetPosition(ctx, randPositionId)
+	if err != nil {
+		return nil, err
 	}
 
-	// pick a random position
-	randPosition := positions[rand.Intn(len(positions))]
+	withdrawAmount := sim.RandomDecAmount(position.Liquidity)
 
-	// get percentage amount from 1 to 100 to withdraw liquidity
-	randPerc := sim.RandomDecAmount(sdk.OneDec())
+	// check if there is enough position liquidity to withdraw
+	availableLiquidity, err := k.GetPositionLiquidity(ctx, randPositionId)
+	if err != nil {
+		return nil, err
+	}
 
-	withdrawAmountInt := randPosition.Liquidity.Mul(randPerc)
+	if withdrawAmount.GT(availableLiquidity) {
+		return nil, fmt.Errorf("insufficient liquidity requested to withdraw.")
+	}
 
 	return &cltypes.MsgWithdrawPosition{
-		PositionId:      randPosition.PositionId,
-		Sender:          sender.Address.String(),
-		LiquidityAmount: withdrawAmountInt,
+		PositionId:      position.PositionId,
+		Sender:          position.GetAddress(),
+		LiquidityAmount: withdrawAmount,
 	}, nil
 }
 
-func RandMsgCollectFees(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgCollectFees, error) {
-	rand := sim.GetRand()
+func RandMsgCollectFeesFullFlow(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgCollectFees, error) {
 	// get random pool
-	_, poolDenoms, err := getRandCLPool(k, sim, ctx)
+	clPool, poolDenoms, err := getRandCLPool(k, sim, ctx)
 	if err != nil {
 		return nil, err
 	}
 
+	poolI, err := k.GetPool(ctx, clPool.GetId())
+	if err != nil {
+		return nil, err
+	}
+
+	// positionCreator creates the position with pool denoms
 	// get random user address with the pool denoms
 	sender, _, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
 	if !senderExists {
 		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
 	}
 
-	positions, err := k.GetUserPositions(ctx, sender.Address, 0)
+	// get random user positions
+	positions, err := k.GetUserPositions(ctx, sender.Address, clPool.GetId())
 	if err != nil {
 		return nil, fmt.Errorf("position does not exist")
 	}
 
-	if len(positions) == 0 {
-		return nil, fmt.Errorf("user does not have any position")
+	var positionIds []uint64
+	for _, position := range positions {
+		positionIds = append(positionIds, position.PositionId)
+	}
+
+	// SwapOwner performs the swap from token0 to token1
+	swapOwner, swapOwnerTokens, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
+	if !senderExists {
+		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
+	}
```

Comment:
Why don't we do the same thing as above with checking positionIds in a pool ID and go from there?

Reply:
because for MsgFeesRequest we need senders position Ids so, i thought it would make more sense to get random position for that specific sender instead of random position in a pool